### PR TITLE
[feat] Parameter sets, part 3, take 2

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Add remaining parameter sets for classes A-H (Addresses - Events)
- Add overload methods using Parameter objects for classes A-H
- Add unit tests, fixtures needed, cassettes for overloaded methods (services)
- Fix bug in parameter dictionary generation that wasn't accounting for non-public properties
  - Adds unit test to verify this works, prevent regression
- Fix bug in parameter dictionary generation that wasn't serializing lists properly
- Minor cleanup, bug fixes of existing parameter set classes as needed

# Testing

- New unit tests for overloaded functions for parameter sets A-H (Addresses - Events)
  - New cassettes

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Comment by nwithan8** [comment]
89.99 doesn't round up to 90. I feel like I'm in college again begging my professor for a grade bump.
image

File: EasyPost/Utilities/Internal/Extensions/Dictionaries.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
+using System;
```
Comment (Justintime50) [inline EasyPost/Utilities/Internal/Extensions/Dictionaries.cs:1]: Thought: The amount of utility code this repo needs is astronomically too large for what the library actually does/is. I'd like us over time to be trimming this down, not adding more to it.

How can we do more with less? What patterns can we identify and refactor (without affecting user expectations and behavior under the hood) to remove dead or duplicated code? Something to think and improve on.

**Review by Justintime50 — APPROVED** [review]
Nice work on the tests, thanks for doing that.

File: EasyPost/Utilities/Internal/Extensions/Dictionaries.cs (hunk 1 above)
Comment (nwithan8) [reply EasyPost/Utilities/Internal/Extensions/Dictionaries.cs:1]: I can start in the next PR by ripping out the three overloads for this method that I added that aren't being used :)

**Review by jchen293 — APPROVED** [review]

**Pull request merged**